### ISSUE TITLE
toReactive nullish initial value

### ISSUE THREAD

**Issue by IMAT13** [vueuse/vueuse#4059]
### Describe the bug

I encountered an error while using the **reactiveComputed** composable in VueUse. After some investigation, I discovered that the **toReactive** composable, used internally by **reactiveComputed**, has a problem with the initial **Nullish** value. I don't know if this behavior is desired or not but here are the details. Should the getter of reactiveComputed always return objects? I think it would be common to get an undefined.

**Steps to Reproduce:**     

Create a **reactiveComputed** composable with a getter that might initially evaluate to **undefined**.
Observe the error that occurs due to the **toReactive** composable not handling the undefined value correctly.

**Expected Behavior:**

The **reactiveComputed** composable should handle initial undefined values and convert them properly using **toReactive**.

**Actual Behavior:**

The initial evaluation of the getter in **reactiveComputed** results in undefined, which causes an issue in the toReactive composable's internal logic.

**Error Message:**

![image](https://github.com/vueuse/vueuse/assets/100535072/050de2dd-516b-470c-bdd9-66d7b83c2037)

**Code Snippet:**
```javascript
<script setup>
import { reactiveComputed } from "@vueuse/core";
let flag = false;
setTimeout(() => {
  flag = true;
}, 100);

const state = reactiveComputed(() => {
  return flag
    ? {
        foo: "bar",
        bar: "baz",
      }
    : undefined;
});
</script>

<template>
  <div>reactiveComputed : {{ state }}</div>
</template>



``` 

### Reproduction

https://stackblitz.com/edit/vitejs-vite-onfwms

### System Info

```Shell
I don't think the system info is significant in this scenario. Besides I get an error running the command that would take forever to solve.
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by isolcat** [comment]
When you initially pass undefined to reactiveComputed, it will throw an error. The documentation mentions that reactiveComputed expects an Object.
![reactiveComputed docs](https://s2.loli.net/2024/06/22/L84Gazj3PrxNX6E.png)
If you want to achieve this, you can do it like this:
```vue
<script setup>
import { reactiveComputed } from '@vueuse/core';
import { ref } from 'vue';

let flag = ref(false);
setTimeout(() => {
  flag.value = true;
}, 2000);

const state = reactiveComputed(() => {
  return flag.value
    ? {
        foo: 'bar',
        bar: 'baz',
      }
    : {};
});
</script>

<template>
  <div>reactiveComputed : {{ state }}</div>
</template>
```

**Comment by IMAT13** [comment]
@isolcat 
yeah, you are right, I have seen it in the document, but I think it's a little bit odd that the getter of computed itself can result in undefined but reactiveComputed can not, it's just the reactive limitation that it only accepts objects or something else. besides the example I have provided is just a modification of the document to replicate the problem, in reality, I don't have access to the getter I'm passing to reactiveComputed.😞

**Comment by ilyaliao** [comment]
In `Vue`, `reactive` can only accept an `object`, and since `reactiveComputed` itself is based on `reactive`, I think this is part of the design. Maybe there will be plans to change it in the future :)
